irohad/consensus/yac/storage/impl/yac_vote_storage.cpp

```diff
@@ -107,6 +112,25 @@ namespace iroha {
         }
       }
 
+      boost::optional<Round> YacVoteStorage::getLastFinalizedRound() const {
+        return last_round_;
+      }
+
+      boost::optional<Answer> YacVoteStorage::getState(
+          const Round &round) const {
+        auto proposal_storage =
+            std::find_if(proposal_storages_.begin(),
+                         proposal_storages_.end(),
+                         [&round](const auto &proposal_state) {
+                           return proposal_state.getStorageKey() == round;
+                         });
```

Comment:
Could `YacVoteStorage::getProposalStorage(const Round &round)` be reused here?
https://github.com/hyperledger/iroha/blob/d0a9e924c4498b89c70f8560587d5fa95429254f/irohad/consensus/yac/storage/impl/yac_vote_storage.cpp#L20-L26

Reply:
Nice catch, thanks

Reply:
But naive reusing impossible here:
`getProposalStorage` method is not const. We want to use it in a const context. Fix declaration of `getProposalStorage` with const we will fail because auto has deduced const iterator which uses in non-const context.

I add ``getProposalStorage` for const version.